rust/core-lib/src/tabs.rs

```diff
@@ -1019,6 +1111,18 @@ impl Documents {
     }
 }
 
+
+/// Returns the modification timestamp for the file at a given path,
+/// if present.
+pub fn get_file_mod_time<P>(path: P) -> Option<SystemTime>
```

Comment:
I'd be curious how resistant all this logic is to races. From my understanding, View.new_view_with_file reads the file, _then_ does set_path which eventually drills down to this call. If the file is changed between these two invocations, I think we can get a situation where it has the old contents but believes the timestamp is fresh.

Getting this right is hard, in such races a lot can go wrong. I'd lean toward not doing anything heroic in this PR, but filing an issue.

Reply:
You're right, we could definitely race here. We could definitely try `flock`ing while we do the read / get the modification date without adding much complexity? Seems reasonable, but would add a dep ([fs2](https://github.com/danburkert/fs2-rs)?). I don't know much about how file locking works in general, possible there's some problem with this approach I'm not aware of.

Reply:
I'm not expert on `flock` and friends, but my understanding is that they give very weak guarantees (my recollection from years back is that they're especially weak on network filesystems, which these days are rarer but not unheard of). I think figuring out the right thing to do is something of an investigation, as I don't think there are any clear, simple wins.

Reply:
Just musing a bit, but couldn't we also just grab the mod time before and after the read, retrying if there's a change?

Reply:
That sounds plausible. Another possibility is registering the watcher before loading, so that a notification would come in.